src/main/java/com/endava/internship/collections/StudentList.java

```diff
@@ -1,145 +1,392 @@
 package com.endava.internship.collections;
 
-import java.util.Collection;
-import java.util.Iterator;
-import java.util.List;
-import java.util.ListIterator;
+import java.util.*;
+import java.util.function.Consumer;
 
-public class StudentList implements List<Student> {
+public class StudentList<T> implements List<T> {
+
+    private static final int DEFAULT_CAPACITY = 10;
+
+    private int size;
+
+    private int capacity = 10;
+
+    private Object[] elements;
+
+    public StudentList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+    }
+
+    public StudentList(int capacity) {
+        this.capacity = capacity;
+        this.elements = new Object[capacity];
+    }
 
     @Override
     public int size() {
-        //TODO
-        return 0;
+        return size;
     }
 
     @Override
     public boolean isEmpty() {
-        //TODO
-        return false;
+        return size == 0;
     }
 
     @Override
-    public boolean contains(Object o) {
-        //TODO
+    public boolean contains(Object t) {
+        if (Objects.isNull(t)) {
+            for (int i = 0; i < size; i++) {
+                if (null == elements[i]) {
+                    return true;
+                }
+            }
+        }else{
+            for (int i = 0; i < size; i++) {
+                if (t.equals(elements[i])) {
+                    return true;
+                }
+            }
+        }
         return false;
     }
 
     @Override
-    public Iterator<Student> iterator() {
-        //TODO
-        return null;
+    public Iterator<T> iterator() {
+        return new StudentList.Iter();
+    }
+    private class Iter implements Iterator<T> {
+        int cursor;       // index of next element to return
+        int lastElem = -1; // index of last element returned; -1 if no such
+
+
+        // prevent creating a synthetic constructor
+        Iter() {}
+
+        public boolean hasNext() {
+            return cursor != size;
+        }
+
+        public T next() {
+            int i = cursor;
+            if (i >= size) {
+                throw new NoSuchElementException();
+            }
+            Object[] elements = StudentList.this.elements;
+
+            if (i >= elements.length)
+                throw new ConcurrentModificationException();
+            cursor = i + 1;
+            return (T) elements[lastElem = i];
+        }
+
+        public void remove() {
+            if (lastElem < 0) {
+                throw new IllegalStateException();
+            }
+            try {
+                StudentList.this.remove(lastElem);
+                cursor = lastElem;
+                lastElem = -1;
+            } catch (IndexOutOfBoundsException ex) {
+                throw new ConcurrentModificationException();
+            }
+        }
+
+        @Override
+        public void forEachRemaining(Consumer<? super T> action) {
```

Comment:
no need to implement this method, you haven't passed lambdas and streams, yet!

Reply:
Okay